src/features/navbar/components/BitwiseNavbar.tsx

```diff
@@ -0,0 +1,191 @@
+import { CustomSelect } from 'common/components/CustomSelect';
+import { useAuth, useLogout } from 'features/auth/hooks';
+import { FC, useEffect, useState } from 'react';
+import Nav from 'react-bootstrap/Nav';
+import { useTranslation } from 'react-i18next';
+import { languages } from '../../../i18n/config';
+import { useNavLinks } from '../hooks/useNavLinks';
+import { CustomNavAction, CustomNavLink } from './CustomNavLink';
+import { Logo } from './Logo';
+import { NavUserDetails } from './NavUserDetails';
+import { ThemeToggle } from '../../themes/ToggleSwitch';
+import { Button, Container, Modal, Navbar, Offcanvas } from 'react-bootstrap';
+import styled from 'styled-components';
+import { useTheme } from 'features/themes/useTheme';
+import light from 'themes/light';
+import dark from 'themes/dark';
+import { environment } from 'environment';
+import { EnvironmentConfiguration } from 'environment/types';
+import { useModal } from 'react-modal-hook';
+import { LoadingButton } from 'common/components/LoadingButton';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+type Props = {
+  closeVerticalNav?: () => void;
+};
+
+type LanguageOption = {
+  label: string;
+  value: string;
+};
+
+export const ResponsiveOffCanvasBody = styled(Offcanvas.Body)`
+  display: flex;
+
+  @media (min-width: 768px) {
+    flex-direction: row;
+    align-items: center;
+  }
+
+  @media (max-width: 767px) {
+    flex-direction: column;
+  }
+`;
+
+export const ResponsiveNavLinks = styled(Nav)`
+  @media (min-width: 768px) {
+    flex-direction: row;
+  }
+
+  @media (max-width: 767px) {
+    flex-direction: column;
+    width: 100%;
+  }
+`;
+
+export const ResponsiveSection = styled(Nav)`
+  @media (min-width: 768px) {
+    justify-content: end;
+    align-items: center;
+    margin-right: 1rem;
+  }
+`;
+
+const getNavBackgroundColor = (isMobilePerspective: boolean, theme: string) => {
+  if (theme === 'light') {
+    if (isMobilePerspective) {
+      return light.nav.vertical.backgroundColor;
+    }
+
+    return light.nav.horizontal.backgroundColor;
+  }
+
+  if (isMobilePerspective) {
+    return dark.nav.vertical.backgroundColor;
+  }
+
+  return dark.nav.horizontal.backgroundColor;
+};
```

Comment:
It would be best if we can keep styling information out of javascript. Especially because this can all be done with css media queries.